cscs-checks/prgenv/src/alloc_speed.cpp

```diff
@@ -0,0 +1,34 @@
+#include <sys/time.h>
+#include <iostream>
+#include <algorithm>
+
+/// Wall-clock time in seconds.
+inline double wtime()
+{
+    timeval t;
+    gettimeofday(&t, NULL);
+    return double(t.tv_sec) + double(t.tv_usec) / 1e6;
+}
+
+double test_alloc(size_t n)
+{
+    double t0 = wtime();
+    /* time to allocate + fill */
+    char* ptr = (char*)std::malloc(n);
+    std::fill(ptr, ptr + n, 0);
+    double t1 = wtime();
+    /* time fo fill */
+    std::fill(ptr, ptr + n, 0);
+    double t2 = wtime();
```

Comment:
can you please add this lines after line 22:
/* harmless: add zero to t0 to prevent full optimization of the code with GCC */
    t0 += ptr[0];

Reply:
This solved the problem. Thanks @toxa81.